components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/ApplicationImportExportManager.java

```diff
@@ -172,18 +180,47 @@ public boolean isOwnerAvailable(String userId) throws APIManagementException {
     /**
      * Check whether a target Tier is available to subscribe
      *
-     * @param targetTier Target Tier
-     * @param api        - {@link API}
+     * @param targetTier     Target Tier
+     * @param apiTypeWrapper - {@link ApiTypeWrapper}
      * @return true, if the target tier is available
      */
-    private boolean isTierAvailable(Tier targetTier, API api) {
-        APIIdentifier apiId = api.getId();
-        Set<Tier> availableTiers = api.getAvailableTiers();
+    private boolean isTierAvailable(Tier targetTier, ApiTypeWrapper apiTypeWrapper) {
+        Set<Tier> availableTiers = null;
+        API api = null;
+        APIProduct apiProduct = null;
+        if (!apiTypeWrapper.isAPIProduct()) {
+            api = apiTypeWrapper.getApi();
+            availableTiers = api.getAvailableTiers();
+        } else {
+            apiProduct = apiTypeWrapper.getApiProduct();
+            availableTiers = apiProduct.getAvailableTiers();
+        }
         if (availableTiers.contains(targetTier)) {
             return true;
         } else {
-            log.error("Tier:" + targetTier.getName() + " is not available for API " + apiId.getApiName() + "-" +
-                    apiId.getVersion());
+            if (!apiTypeWrapper.isAPIProduct()) {
+                log.error("Tier:" + targetTier.getName() + " is not available for API " + api.getId().getApiName() + "-" +
+                        api.getId().getVersion());
+            } else {
+                log.error("Tier:" + targetTier.getName() + " is not available for API Product " + apiProduct.getId().getName() + "-" +
+                        apiProduct.getId().getVersion());
+            }
+            return false;
+        }
+    }
+
+    /**
+     * Check whether the object is a type of ApiProduct
+     *
+     * @param object        - {@link Object}
+     * @return true, if the object is an ApiProduct, otherwise false
+     */
+    private boolean isApiProduct(Object object) {
+        try {
+            //Cast object to ApiProduct
+            APIProduct apiProduct = (APIProduct) object;
+            return (apiProduct != null) ? true : false;
+        } catch (Exception e) {
             return false;
```

Comment:
can't we use instanceOf?

Also, don't catch generic "Exception"s.

Reply:
Thanks for the input. Fixed the requested changed